forc-pkg/src/manifest.rs

```diff
@@ -127,6 +128,52 @@ impl Manifest {
     }
 }
 
+// Checks for unused manifest keys and displays a warning for each key present.
+//
+// This strategy is a combination of [cargo/util/toml/mod.rs](https://github.com/rust-lang/cargo/blob/489b66f2e458404a10d7824194d3ded94bc1f4e4/src/cargo/util/toml/mod.rs#L100),
+// [serde_ignored](https://docs.rs/serde_ignored/latest/serde_ignored/)'s default example and `forc-pkg::pkg::validate`.
+fn warn_unused_key(manifest: &str) -> anyhow::Result<()> {
+    let manifest_str = manifest;
+    let toml = &mut toml::de::Deserializer::new(manifest_str);
+    let mut unused_keys = BTreeSet::new();
+    let _manifest: Manifest = serde_ignored::deserialize(toml, |path| {
+        let mut key = String::new();
+        verify_keypath(&mut key, &path);
```

Comment:
Judging by the  `serde_ignored` docs example it looks like we should be able to do:
```rust
unused_keys.insert(path.to_string());
```
here. Could you explain a little further why we need the `verify_keypath` function?

Reply:
To be honest, I chose to keep the majority of the code from the `Cargo` example, the function was necessary for the loop to work with the logic that was there. Do you think using the `serde_ignored` docs method is cleaner?

Reply:
The docs example removes the need for the entire `verify_keypath` function, so certainly cleaner :) Does it still result in nice warnings? Would you mind pasting an example of what the warning looks like in both cases here?